### PR TITLE
[BD-5] BB-2504: Add grade status summary page. [WIP]

### PR DESCRIPTION
**Pull request by ahmedbilal**
Add a status message on the “Your grade” step explaining how the final grade is calculated/determined.

**JIRA tickets**: 
* https://openedx.atlassian.net/browse/TNL-7274
* https://openedx.atlassian.net/browse/OSPR-4624

**Dependencies**: None

**Screenshots**: Always include screenshots if there is any change to the UI.

**Merge deadline**: None

**Testing instructions**:

1. Create a sample course.
2. Create ORA (Open Response Assessment) one for each assessment type e.g **self**, **peer**, **staff**, **self** + **peer**, ....., etc.
3. Create student and enroll them into the created course.
4. Get them to attempt the ORA assessments and get them to grading page.
5. Check the output HTML whether the grade explanation message appears or not.

**Author notes and concerns**:

1. There is some strings missing and for them I have added pseudo strings. I will replace them with actual ones once I get them from upstream.

**Reviewers**
- [ ] @giovannicimolin
- [ ] edX reviewer[s] TBD

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @ahmedbilal! I've created [OSPR-4624](https://openedx.atlassian.net/browse/OSPR-4624) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

File: openassessment/xblock/grade_mixin.py [hunk 1]
```diff
@@ -130,6 +130,49 @@ def render_grade_complete(self, workflow):
         # when all the criteria in the rubric are feedback-only (no options).
         score = workflow['score']
 
+        first_sentence = "You have successfully completed this problem and received a {}/{}.".format(
+            score["points_earned"], score["points_possible"]
+        )
+        second_sentence = ""
```
Comment (giovannicimolin) [inline openassessment/xblock/grade_mixin.py:136]: Can you move this block of code to a separate function and pass `assessment_steps` as an argument?
Something like `_get_score_explanation()`.

File: openassessment/xblock/grade_mixin.py [hunk 2]
```diff
@@ -130,6 +130,49 @@ def render_grade_complete(self, workflow):
...
+
+        elif is_staff_assessment and is_peer_assessment and not is_self_assessment:
+            # TODO: Sentence 2 not mentioned yet by the reporter. The following would be replaced
+            #       once I get the actual sentence
+            second_sentence = "This is staff + peer assessment and Sentence 2 isn't decided yet"
+
+        elif is_self_assessment and is_staff_assessment and is_peer_assessment:
+            second_sentence = "The grade for this problem is determined by your Staff Grade."
+
+        score["explanation"] = first_sentence + " " + second_sentence
```
Comment (giovannicimolin) [inline openassessment/xblock/grade_mixin.py:174]: Use either `.format` for formatting strings.

Comment (giovannicimolin) [inline openassessment/xblock/grade_mixin.py:174]: `score` is directly returned from the Workflow API, don't modify it. Instead add a new variable to the context bellow.

**Comment by giovannicimolin** [comment]
@ahmedbilal Can you rebase the PR and fix the conflicts?

**Comment by ahmedbilal** [comment]
> @ahmedbilal Can you rebase the PR and fix the conflicts?
@giovannicimolin 
I rebased it on edx:edx-ora2:master but it add a lot of other changes into Files changed. Is it expected or did I do something wrong?

**Comment by 0x29a** [comment]
Hey @ahmedbilal.

I see your build is also failing because of exception in `AssessmentSerializerTest.test_full_assessment_dict_criteria_no_options` test.

I have the same problem: https://github.com/edx/edx-ora2/pull/1425#issuecomment-639333351

**Comment by ahmedbilal** [comment]
> Hey @ahmedbilal.
> 
> I see your build is also failing because of exception in `AssessmentSerializerTest.test_full_assessment_dict_criteria_no_options` test.
> 
> I have the same problem: [#1425 (comment)](https://github.com/edx/edx-ora2/pull/1425#issuecomment-639333351)

Thanks :+1:

**Comment by edx-webhook** [comment]
@ahmedbilal Even though your pull request wasn’t merged, please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/edx-ora2&entry.867055334=https://github.com/edx/edx-ora2/pull/1423&entry.1484655318=https://github.com/ahmedbilal&entry.752974735=2020-06-02+08:00&entry.1917517419=2020-06-06+06:27&entry.2133058324=No) so we can improve your experience in the future.

**Pull request closed without merging**